lib/fencing/st_client.c

```diff
@@ -823,11 +829,12 @@ stonith_action_execute_async(stonith_action_t * action,
 
     action->userdata = userdata;
     action->done_cb = done;
+    action->fork_cb = fork_cb;
     action->async = 1;
 
     rc = internal_stonith_action_execute(action);
 
-    return rc < 0 ? rc : action->pid;
+    return rc < 0 ? rc : 0;
```

Comment:
It looks like this could now simplify to return internal_stonith_action_execute(action);